lib/Core/BuildEngine.cpp

```diff
@@ -492,10 +492,17 @@ class BuildEngineImpl : public BuildDBDelegate {
 
     // Create the task for this rule.
     Task* task = ruleInfo.rule.action(buildEngine);
+    assert(task && "rule action returned null task");
 
     // Find the task info for this task.
     auto taskInfo = getTaskInfo(task);
     assert(taskInfo && "rule action returned an unregistered task");
+
+    // Same as the above assert, but for catching this in production as a
+    // clearer cause for demandRule crashes (maybe?)
+    if (!taskInfo)
+      abort();
+
```

Comment:
I would drop this... I think we will be able to tell this is true from the exact crash anyway...